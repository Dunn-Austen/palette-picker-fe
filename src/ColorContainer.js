import React, {useState, useEffect} from 'react';
import ColorCard from './ColorCard';
import './ColorContainer.css';

const ColorContainer = ({paletteColors, isSavedStatuses, setIsSavedFunctions}) => {
  let {paletteColor_1, paletteColor_2, paletteColor_3, paletteColor_4, paletteColor_5} = paletteColors;
  let {isSaved_1, isSaved_2, isSaved_3, isSaved_4, isSaved_5} = isSavedStatuses;
  let {setIsSaved_1, setIsSaved_2, setIsSaved_3, setIsSaved_4, setIsSaved_5} = setIsSavedFunctions;
  let colors = [paletteColor_1, paletteColor_2, paletteColor_3, paletteColor_4, paletteColor_5];

  const displayColorCards = colors.map((color, index) => {
    return (
      <ColorCard
        hexCode={color}
        key={color}
      />
    );
  });

  return (
    <section className='color-container'>
      {displayColorCards}
    </section>
  );
}

export default ColorContainer;
