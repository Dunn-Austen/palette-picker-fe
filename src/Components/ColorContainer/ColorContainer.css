.color-container {
  align-items: center;
  border: 5px ridge white;
  box-sizing: border-box;
  display: flex;
  flex-direction: row;
  height: 80%;
  justify-content: center;
  width: 70%;
}
