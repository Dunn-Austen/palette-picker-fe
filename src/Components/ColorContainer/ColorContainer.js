import React from 'react';
import ColorCard from '../ColorCard/ColorCard';
import './ColorContainer.css';
import PropTypes from 'prop-types';

const ColorContainer = ({paletteColors, isSavedStatuses, setIsSavedFunctions}) => {
  let { paletteColor_1, paletteColor_2, paletteColor_3, paletteColor_4, paletteColor_5 } = paletteColors;
  let { isSaved_1, isSaved_2, isSaved_3, isSaved_4, isSaved_5 } = isSavedStatuses;
  let { setIsSaved_1, setIsSaved_2, setIsSaved_3, setIsSaved_4, setIsSaved_5 } = setIsSavedFunctions;
  let combinedColorData =
    [
      {hexCode: paletteColor_1, isSaved: isSaved_1, setIsSaved: setIsSaved_1},
      {hexCode: paletteColor_2, isSaved: isSaved_2, setIsSaved: setIsSaved_2},
      {hexCode: paletteColor_3, isSaved: isSaved_3, setIsSaved: setIsSaved_3},
      {hexCode: paletteColor_4, isSaved: isSaved_4, setIsSaved: setIsSaved_4},
      {hexCode: paletteColor_5, isSaved: isSaved_5, setIsSaved: setIsSaved_5},
    ];

  const displayColorCards = combinedColorData.map((colorData, index) => {
    return (
      <ColorCard
        hexCode={colorData.hexCode}
        isSaved={colorData.isSaved}
        setIsSaved={colorData.setIsSaved}
        key={index}
      />
    );
  });

  return (
    <section className='color-container'>
      {displayColorCards}
    </section>
  );
}

export default ColorContainer;

ColorContainer.propTypes = {
  paletteColors: PropTypes.object,
  isSavedStatuses: PropTypes.object,
  setIsSavedFunctions: PropTypes.object
}
