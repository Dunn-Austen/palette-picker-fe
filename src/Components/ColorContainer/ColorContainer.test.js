import React from 'react';
import { shallow } from 'enzyme';
import ColorContainer from './ColorContainer';

describe('App', () => {
  it('should match the snapshot', () => {
    const mockPaletteColors = {
      paletteColor_1: '#444443',
      paletteColor_2: '#444443',
      paletteColor_3: '#444443',
      paletteColor_4: '#444443',
      paletteColor_5: '#444443'
    };
    const mockIsSavedStatuses = {
      isSaved_1: false,
      isSaved_2: false,
      isSaved_3: false,
      isSaved_4: false,
      isSaved_5: false
    };
    const mockSetIsSavedFunctions= jest.fn()
    const wrapper = shallow(<ColorContainer 
      paletteColors={mockPaletteColors}
      isSavedStatuses = {mockIsSavedStatuses}
      setIsSavedFunctions={mockSetIsSavedFunctions}
    />);

    expect(wrapper).toMatchSnapshot();
  });
})
