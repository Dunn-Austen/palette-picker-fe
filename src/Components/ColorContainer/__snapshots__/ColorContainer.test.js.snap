// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<section
  className="color-container"
>
  <ColorCard
    hexCode="#444443"
    isSaved={false}
    key="0"
  />
  <ColorCard
    hexCode="#444443"
    isSaved={false}
    key="1"
  />
  <ColorCard
    hexCode="#444443"
    isSaved={false}
    key="2"
  />
  <ColorCard
    hexCode="#444443"
    isSaved={false}
    key="3"
  />
  <ColorCard
    hexCode="#444443"
    isSaved={false}
    key="4"
  />
</section>
`;
